docs/Doxyfile.template

```diff
@@ -0,0 +1,2708 @@
+# Doxyfile 1.9.4
+
+# This file is a template for the Doxyfile used to generate Flutter
```

Comment:
Is this based on the doxyfile at https://github.com/flutter/engine/blob/main/Doxyfile or does it start from scratch. Notably, that file is already setup for docset generation which comes in handy.

Reply:
I based it on that Doxyfile.  Some things are changed (notably, I turned on the directory sharding so that there aren't thousands of files in one dir).

I was wondering what that Doxyfile was actually used for, though, since it doesn't appear to be published anywhere.